webapi/src/main/scala/org/knora/webapi/store/iiif/SipiConnector.scala

```diff
@@ -321,9 +321,14 @@ class SipiConnector extends Actor with ActorLogging {
         // ask Sipi to return the XSL transformation
         val request = new HttpGet(xsltFileUrl)
 
-        for {
+        val sipiResponseTry: Try[SipiGetTextFileResponse] = for {
             responseStr <- doSipiRequest(request)
         } yield SipiGetTextFileResponse(responseStr)
+
+        sipiResponseTry.recover {
+            case badRequestException: BadRequestException => throw SipiException(s"Unable to get XSL transformation file $xsltFileUrl from Sipi: ${badRequestException.message}")
```

Comment:
`sipiGetXsltTransformationRequestV2` is called in two cases: for XSLT used by `StandoffResponderV2` to turn XML into HTML and for XSLT used the TEI transformer in `ResourcesResponderV2`. So I guess testing it with the TEI route is fine.

However, `SipiGetTextFileRequest` is as the message sent to `SipiConnector` that then internally calls `sipiGetXsltTransformationRequestV2` seems strange to me. Shouldn't either the message be more specific or the method more generic?

Reply:
In the future, Knora might require other text files from Sipi than XSL transformations and then the error message thrown by `sipiGetXsltTransformationRequestV2 ` might be misleading.

Reply:
Quite true, I'll fix that.

Reply:
Done in 080dd44.